nncf/common/tensor_statistics/statistic_point.py

```diff
@@ -71,6 +71,26 @@ def add_statistic_point(self, statistic_point: StatisticPoint) -> None:
 
             self.data[target_node_name].append(statistic_point)
 
+    def remove_statistic_points(self, algorithm_name: str) -> None:
+        """
+        Method to remove statistic point associated with a given algorithm
+        from statistic point container.
+
+        :param algorithm: Algorithm name
+        """
+        data_copy = dict(self.data)
+
+        for target_node_name, statistic_points in data_copy.items():
+            # Reassign every target node name IF it doesn't contain the given algorithm
+            self.data[target_node_name] = [
+                _statistic_point
+                for _statistic_point in statistic_points
+                if algorithm_name not in _statistic_point.algorithm_to_tensor_collectors
+            ]
+
+            if not self.data[target_node_name]:
+                del self.data[target_node_name]
+
```

Comment:
After the internal discussion, I suggest improving the current proposal with something like this:
```suggestion
    def remove_statistic_points(self, algorithm_key: str) -> None:
        """
        Method to remove statistic point associated with a given algorithm
        from statistic point container.

        :param algorithm_key: Algorithm key.
        """
        reduced_data = {}

        for target_node_name, statistic_points in self.data.items():
            # Reassign every target node name IF it doesn't contain the given algorithm
            reduced_statistics = []
            for statistic_point in statistic_points:
            """
            ...
            Collect the new reduced_statistics here
            ...
            """
            if reduced_statistics:
                reduced_data[target_node_name] = reduced_statistics

        self.data = reduced_data
```
And do the measure of overall consumption using `quantize` with the option `fast_bias_correction=False`, to enable the most memory consumption algorithm.